### PR TITLE
pixi-pack: init at 0.3.1

### PR DESCRIPTION
**Pull request by xiaoxiangmoe**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pi/pixi-pack/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    rev = "refs/tags/v${version}";
+    hash = "sha256-gzP/01pck14cMN0PopoCO27dxncWz4yIZFAzXU+4IQ0=";
+  };
+  cargoHash = "sha256-2kr4mFqG8IuRR3XiiGFlEX70c4NIcdF0NICwyrdjhcs=";
+
+  # As the version is updated, the number of failed tests continues to grow.
+  doCheck = false;
+
+  passthru.updateScript = nix-update-script { };
+  passthru.tests.version = testers.testVersion {
```
Comment (lucasew) [inline pkgs/by-name/pi/pixi-pack/package.nix:27]: Can you use versionCheckHook instead?

File: pkgs/by-name/pi/pixi-pack/package.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    repo = "pixi-pack";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-gzP/01pck14cMN0PopoCO27dxncWz4yIZFAzXU+4IQ0=";
+  };
+  cargoHash = "sha256-2kr4mFqG8IuRR3XiiGFlEX70c4NIcdF0NICwyrdjhcs=";
+
+  # As the version is updated, the number of failed tests continues to grow.
+  doCheck = false;
+
+  passthru.updateScript = nix-update-script { };
```
Comment (lucasew) [inline pkgs/by-name/pi/pixi-pack/package.nix:26]: Put this right before meta

File: pkgs/by-name/pi/pixi-pack/package.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "pixi-pack";
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "Quantco";
+    repo = "pixi-pack";
+    rev = "refs/tags/v${version}";
```
Comment (lucasew) [inline pkgs/by-name/pi/pixi-pack/package.nix:18]: use tag instead of rev

Comment (xiaoxiangmoe) [reply pkgs/by-name/pi/pixi-pack/package.nix:18]: tag can't using nix-update-script now.

**Comment by lucasew** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 369397`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>gnuk</li>
  </ul>
</details>
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>affine</li>
    <li>ananicy</li>
    <li>libretranslate (python312Packages.libretranslate)</li>
    <li>libretranslate.dist (python312Packages.libretranslate.dist)</li>
    <li>nanoboyadvance</li>
    <li>pixi-pack</li>
    <li>python312Packages.ltpycld2</li>
    <li>python312Packages.ltpycld2.dist</li>
    <li>python313Packages.ltpycld2</li>
    <li>python313Packages.ltpycld2.dist</li>
    <li>schedtool</li>
    <li>spike</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 11 packages built:</summary>
  <ul>
    <li>armTrustedFirmwareTools</li>
    <li>bonsai</li>
    <li>dnsdbq</li>
    <li>fusee-launcher</li>
    <li>hareThirdParty.hare-ev</li>
    <li>klipper-firmware</li>
    <li>klipper-flash</li>
    <li>nitrokey-trng-rs232-firmware</li>
    <li>qmk</li>
    <li>qmk.dist</li>
    <li>simavr</li>
  </ul>
</details>

**Comment by xiaoxiangmoe** [comment]
@lucasew I tested it in x86_64-darwin and it seems works well.

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

File: pkgs/by-name/pi/pixi-pack/package.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
```
[the remark this answers is not in the document]
Comment (xiaoxiangmoe) [reply pkgs/by-name/pi/pixi-pack/package.nix:18]: No, I can't use it until `https://github.com/Mic92/nix-update/pull/312` is merged

[the remark this answers is not in the document]
Comment (xiaoxiangmoe) [reply pkgs/by-name/pi/pixi-pack/package.nix:19]: Okay, I change it to tag now.

File: pkgs/by-name/pi/pixi-pack/package.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  versionCheckHook,
+  pkg-config,
+  openssl,
+  nix-update-script,
+  pixi-pack,
```
[the remark this answers is not in the document]
Comment (xiaoxiangmoe) [reply pkgs/by-name/pi/pixi-pack/package.nix]: I change it to order of apparition now, except `lib` in first order although `lib` is used in end.

[the remark this answers is not in the document]
Comment (xiaoxiangmoe) [reply pkgs/by-name/pi/pixi-pack/package.nix]: Should I also move `lib` to end?

[the remark this answers is not in the document]
Comment (xiaoxiangmoe) [reply pkgs/by-name/pi/pixi-pack/package.nix]: Is there any lint for parameters order? I may forget these sometimes.

**Comment by xiaoxiangmoe** [comment]
I rebase origin/master now

**Comment by xiaoxiangmoe** [comment]
@GaetanLepage Thank you very much to help me polish this PR.

**Review by GaetanLepage — APPROVED** [review]
LGTM !
Thanks for your patience :)

**Pull request merged**